dashboard/test/models/user_test.rb

```diff
@@ -1046,6 +1046,61 @@ def complete_script_for_user(user, script, completed_date = Time.now)
     assert @student.can_edit_email?
   end
 
+  test 'teacher_managed_account? is false for teacher' do
+    refute @teacher.teacher_managed_account?
+  end
+
+  test 'teacher_managed_account? is false for normal student account with hashed email and password' do
+    refute @student.teacher_managed_account?
+  end
+
+  test 'teacher_managed_account? is false for student account in section with oauth connection' do
+    student_with_oauth = create(:student, encrypted_password: nil, provider: 'facebook', uid: '1111111')
+
+    # join picture section
+    picture_section = create(:section, login_type: Section::LOGIN_TYPE_PICTURE)
+    create(:follower, student_user: student_with_oauth, section: picture_section)
+    student_with_oauth.reload
+    refute student_with_oauth.teacher_managed_account?
```

Comment:
Yes, this is deliberate because I want to make sure that if a student *is* in a section, then even if they don't have a password, it's still not considered teacher-managed if there's an oauth connection. The condition I'd initially used in this PR would have missed this case.